### ISSUE TITLE
Repeated calls to LanguageModel require explicit garbage collection

### ISSUE THREAD

**Issue by keltin13** [ndif-team/nnsight#538]
I am struggling with nnsight tracing using lots of GPU memory over repeated traces of a model. The issue is significantly worse when gradients are involved. Even without gradients, here is one example:

```
import torch
from nnsight import LanguageModel

model = LanguageModel("google/gemma-2-2b", torch_dtype="float16", device_map="cuda:0", dispatch=True)
for param in model.model.parameters():
    param.requires_grad = False

dummy_input = torch.ones((20, 25), dtype=torch.long, device="cuda:0")

def print_mem(name):
    allocated = torch.cuda.memory_allocated() / 1024**3
    reserved = torch.cuda.memory_reserved() / 1024**3
    print(f"{name}: {allocated:.2f} GB allocated, {reserved:.2f} GB reserved")

print_mem("Initial memory")

for i in range(200):
    with model.trace(dummy_input) as tracer:
        _ = model.lm_head.output.save()

    if i % 10 == 0:
        print_mem(f"Step {i}")

print_mem("Final memory")
```

```
Initial memory: 4.87 GB allocated, 5.06 GB reserved
You're using a GemmaTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.
The 'batch_size' argument of HybridCache is deprecated and will be removed in v4.49. Use the more precisely named 'max_batch_size' argument instead.
The 'batch_size' attribute of HybridCache is deprecated and will be removed in v4.49. Use the more precisely named 'self.max_batch_size' attribute instead.
Step 0: 5.12 GB allocated, 5.90 GB reserved
Step 10: 6.79 GB allocated, 7.58 GB reserved
Step 20: 6.79 GB allocated, 8.30 GB reserved
Step 30: 7.98 GB allocated, 8.78 GB reserved
Step 40: 6.31 GB allocated, 9.02 GB reserved
Step 50: 7.50 GB allocated, 9.02 GB reserved
Step 60: 7.50 GB allocated, 9.02 GB reserved
Step 70: 8.69 GB allocated, 9.50 GB reserved
Step 80: 8.69 GB allocated, 9.98 GB reserved
Step 90: 9.89 GB allocated, 10.70 GB reserved
Step 100: 9.89 GB allocated, 10.94 GB reserved
Step 110: 5.83 GB allocated, 11.66 GB reserved
Step 120: 7.03 GB allocated, 11.66 GB reserved
Step 130: 7.03 GB allocated, 11.66 GB reserved
Step 140: 8.22 GB allocated, 11.66 GB reserved
Step 150: 8.22 GB allocated, 11.66 GB reserved
Step 160: 9.41 GB allocated, 11.66 GB reserved
Step 170: 9.41 GB allocated, 11.66 GB reserved
Step 180: 10.60 GB allocated, 11.66 GB reserved
Step 190: 6.55 GB allocated, 12.14 GB reserved
Final memory: 7.50 GB allocated, 12.14 GB reserved
```

You can see that periodically the allocated memory drops down (step 100-110, 180-190) (presumably when garbage collection runs). Adding `gc.collect()` to each loop iteration solves the issue:

```
Step 0: 5.12 GB allocated, 5.90 GB reserved
Step 5: 5.59 GB allocated, 6.38 GB reserved
Step 10: 5.59 GB allocated, 6.38 GB reserved
Step 15: 5.59 GB allocated, 6.38 GB reserved
Step 20: 5.59 GB allocated, 6.38 GB reserved
...
```

`torch.cuda.empty_cache()` alone does not help with the allocated memory, though it does reduce the reserved memory.

Is there any way to address this on the nnsight side rather than manually calling `gc.collect()`. Thanks!
